modules/task/execution_test.go

```diff
@@ -199,3 +200,10 @@ func Complex(car Car) error {
 	}
 	return nil
 }
+
+func TestCastToError(t *testing.T) {
+	s := make(map[string]string)
+	err := castToError(reflect.ValueOf(s))
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "be error but found")
```

Comment:
We decided not to be overly strict about error string comparisons. Using EqualError would mean I need the entire string from the error. That just makes it fragile when we change the wordig etc.